src/plugins/flow.js

```diff
@@ -505,6 +505,8 @@ pp.flowParsePrimaryType = function () {
         node.returnType = this.flowParseType();
 
         return this.finishNode(node, "FunctionTypeAnnotation");
+      } else {
+        this.unexpected();
```

Comment:
This is just minor and makes not a big difference, but maybe adding here just a `break` and then it would call `this.unexpected()` on line 395